packages/static-hosting/lib/static-hosting.ts

```diff
@@ -1,144 +1,99 @@
   enableCloudFrontAccessLogging?: boolean;
-  enableS3AccessLogging?: boolean;
-  zoneName?: string;
-  /**
-   * Used to add Custom origins and behaviors
-   */
-  customOriginConfigs?: Array<SourceConfiguration>;
-  /**
-   * Used to prepend the behaviours introduced as part of customOriginConfigs.
-   * This is a work around until this construct is updated to use the new
-   * Distribution API.
-   * https://docs.aws.amazon.com/cdk/api/latest/docs/aws-cloudfront-readme.html#distribution-api
-   */
-  prependCustomOriginBehaviours?: boolean;
-
-  /**
-   * Optional set of behaviors to override the default behavior defined in this construct
-   */
-  behaviors?: Array<Behavior>;
-  enableErrorConfig?: boolean;
-  enableStaticFileRemap?: boolean;
-  remapPaths?: remapPath[];
-  backendHost?: string;
-  remapBackendPaths?: remapPath[];
+  errorResponsePagePath?: string;
+  webAclId?: string;
   defaultRootObject?: string;
-  enforceSSL?: boolean;
-  /**
-   * Disable the use of the CSP header. Default value is false.
-   */
-  disableCSP?: boolean;
-  /**
-   * AWS limits the max header size to 1kb, this is too small for complex csp headers.
-   * The main purpose of this csp header is to provide a method of setting a report-uri.
-   */
+  certificateArn: string;
+  defaultBehaviorEdgeLambdas: EdgeLambda[];
+  additionalBehaviors?: Record<string, BehaviorOptions>;
+  enableErrorConfig?: boolean;
+  createDnsRecord?: boolean;
+  zoneName?: string;
   csp?: CSP;
-  /**
-   * This will generate a csp based *purely* on the provided csp object.
-   * Therefore disabling the automatic adding of common use-case properties.
-   */
   explicitCSP?: boolean;
+  responseHeadersPolicies?: ResponseHeaderMappings;
 
   /**
-   * Extend the default props for S3 bucket
-   */
-  s3ExtendedProps?: BucketProps;
-
-  /**
-   * Optional WAF ARN
-   */
-  webAclArn?: string;
-
-  /**
-   * Response Header policies
+   * After switching constructs, you need to maintain the same logical ID
+   * for the underlying CfnDistribution if you wish to avoid the deletion
+   * and recreation of your distribution.
+   *
+   * To do this, use escape hatches to override the logical ID created by
+   * the new Distribution construct with the logical ID created by the
+   * old construct
+   *
+   * https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront-readme.html#migrating-from-the-original-cloudfrontwebdistribution-to-the-newer-distribution-construct.
    */
-  responseHeadersPolicies?: ResponseHeaderMappings[];
-}
-
-interface remapPath {
-  from: string;
-  to: string;
+  overrideLogicalId?: string;
 }
 
 export interface ResponseHeaderMappings {
-  header: ResponseHeadersPolicy;
-  pathPatterns: string[];
-  attachToDefault?: boolean;
+  defaultBehaviorResponseHeaderPolicy?: ResponseHeadersPolicy;
+  additionalBehaviorResponsePolicy?: Record<string, ResponseHeadersPolicy>;
 }
 
-export class StaticHosting extends Construct {
-  private staticFiles = [
```

Comment:
Has this functionality been removed entirely?

Reply:
Yeah, because behaviors are to be added outside of static hosting but I'm investigating alternative options to keep this in. It will have to be part of a future PR

Reply:
That would be great. This is vital to our hosting infra so definitely needed!

Reply:
After playing around, we may not require this. The only reason we have this is we create them as additional behaviours so that these behaviours can be excluded from having any lambda association. This was a workaround / hack because of the one lambda limitation.

This is a solved problem now since we just let the function decide whether assets need to pass through or not.